src/app/modules/calendar/calendar-year/calendar-year.component.ts

```diff
@@ -76,10 +77,26 @@ export class CalendarYearComponent implements OnInit {
     this.calendar = this.calendarService.createCalendarForYear();
   }
 
-  handleSickDayClick(day: CalendarItem) {
+  handleAbsenceDayClick(day: CalendarItem) {
+    const isUpdating = !!day.absence;
+
+    const absence: Absence = isUpdating
+      ? day.absence as Absence
+      : {
+        id: -1,
```

Comment:
-1 looks like magic number. Could you please add some explanation?

Reply:
If it's an edit of an absence, we send it, if you add a new one by clicking on the day, we send a fake absence to fill in the fields on the same day.

I'll add this as a comment, before I wrote a bigger code there that was easier to understand, but it was too big